components/pedigree/resources/src/main/resources/pedigree/familyData.js

```diff
@@ -33,8 +33,12 @@ define([
                 this.familyMembersIndex[this.familyMembers[i].id] = i;
             }
 
-            this.warningMessage = familyJSON.hasOwnProperty("warning") ?
-                                  familyJSON.warning : "";
+            this.warningMessage = null;
+            if (familyJSON.hasOwnProperty("contains_sensitive_data") && familyJSON["contains_sensitive_data"]) {
+                this.warningMessage = (familyJSON.hasOwnProperty("sensitive_data_message") && familyJSON["sensitive_data_message"] != "")
+                                      ? familyJSON["sensitive_data_message"]
+                                      : "This pedigree was marked as containing sensitive information. No further details were provided.";
```

Comment:
Defining texts as properties of the class rather than in code will make the pedigree application easier to translate in the future. I suggest to add
```
DEFAULT_WARNING_MESSAGE : "This pedigree was marked as containing sensitive information. No further details were provided.",
```
just before like 14.